lib/decompress/zstd_decompress_block.c

```diff
@@ -713,14 +794,15 @@ HINT_INLINE void ZSTD_overlapCopy8(BYTE** op, BYTE const** ip, size_t offset) {
  *         - ZSTD_overlap_src_before_dst: The src and dst may overlap and may be any distance apart.
  *           The src buffer must be before the dst buffer.
```

Comment:
Complete the documentation. Apparently, `ZSTD_overlap_dst_before_src` is a new case this function covers.